legion/engine/core/ecs/registry.cpp

```diff
@@ -3,6 +3,24 @@
 
 namespace legion::core::ecs
 {
+    component_pool_base* Registry::tryEmplaceFamily(id_type id, std::unique_ptr<component_pool_base>&& ptr)
+    {
+        auto& families = getFamilies();
+        if (families.count(id)) // Check and fetch in order to avoid a possibly unnecessary allocation and deletion.
+            return families.at(id).get();
+
+        // Allocate and emplace if no item was found.
+        return families.emplace(
+            id,
+            std::move(ptr)
+        ).first->second.get(); // std::pair<iterator, bool>.first --> iterator<std::pair<key, value>>->second --> std::unique_ptr.get() --> component_pool_base*
```

Comment:
remove commented code

Reply:
this is not commented out code, it's a comment saying which types are involved. It can get a bit confusing with such layered types